test/functional/test_framework/test_node.py

```diff
@@ -771,6 +773,8 @@ def addconnection_callback(address, port):
             p2p_conn.wait_for_connect()
             self.p2ps.append(p2p_conn)
 
+            if supports_v2_p2p:
+                p2p_conn.wait_until(lambda: p2p_conn.v2_state.tried_v2_handshake)
```

Comment:
You are right, it's not strictly needed in `add_outbound_p2p_connection`, (though it is in `add_p2p_connection`, where the corresponding wait depends on the `send_version` option which may be unset).
I think it makes sense conceptually because it kind of decouples the v2 logic from the message logic. If we'd introduce a similar parameter `send_version`  also for `add_outbound_p2p_connection` (e.g. because we might want to craft a custom version message like `p2p_sendtxrcncl.py` does) we'd need it.